src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X500NameEncoder.cs

```diff
@@ -567,5 +582,70 @@ private static int ExtractValue(ReadOnlySpan<char> chars, Span<char> destination
 
             return written;
         }
+
+        private static Dictionary<string, EncodingRules> CreateEncodingRulesLookup()
+        {
+            const int LookupDictionarySize = 27;
+            Dictionary<string, EncodingRules> lookup = new(LookupDictionarySize, StringComparer.Ordinal)
+            {
+                { Oids.EmailAddress, EncodingRules.IA5String },
+                { Oids.CommonName, EncodingRules.DirectoryString },
+                { Oids.Surname, EncodingRules.DirectoryString },
+                { Oids.GivenName, EncodingRules.DirectoryString },
+                { Oids.Initials, EncodingRules.DirectoryString },
+                { Oids.GenerationQualifier, EncodingRules.DirectoryString },
+                { Oids.DnQualifier, EncodingRules.PrintableString },
+                { Oids.SerialNumber, EncodingRules.PrintableString },
+                { Oids.Pseudonym, EncodingRules.DirectoryString },
+                { Oids.DnsName, EncodingRules.UTF8String },
+                { Oids.IntEmail, EncodingRules.UTF8String },
+                { Oids.Jid, EncodingRules.UTF8String },
+                { Oids.CountryOrRegionName, EncodingRules.PrintableString },
+                { Oids.CountryOrRegionName3C, EncodingRules.PrintableString },
+                { Oids.CountryOrRegionName3N, EncodingRules.NumericString },
```

Comment:
The lengths of these are supposed to be exactly 2, 3, 3 (respectively).  Do we care?

Reply:
I don't think so, because Windows doesn't care. Windows is perfectly happy with:

```C#
X500DistinguishedName dn = new("C=Aglarond");
```